File: algorithms/11_golang/sorting/selection_sort.go
package main

import "fmt"

func selectionSort(a []int) {
	for i := 0; i < len(a)-1; i++ {
		minIdx := i
		for j := i + 1; j < len(a); j++ {
			if a[j] < a[minIdx] {
				minIdx = j
			}
		}
		a[i], a[minIdx] = a[minIdx], a[i]
	}
}

func main() {
	data := []int{33, 7, 91, 12, 5, 5, 78, 2, 44, 19}
	selectionSort(data)
	for i := 1; i < len(data); i++ {
		if data[i-1] > data[i] {
			panic("not sorted")
		}
	}
	fmt.Println("[Go SelectionSort] Selection sort verified:", data)
}

File: algorithms/13_swift/sorting/SelectionSort.swift
func selectionSort<T: Comparable>(_ arr: [T]) -> [T] {
    var a = arr
    for i in 0..<a.count - 1 {
        var minIdx = i
        for j in i + 1..<a.count {
            if a[j] < a[minIdx] { minIdx = j }
        }
        a.swapAt(i, minIdx)
    }
    return a
}

let data = [33, 7, 91, 12, 5, 5, 78, 2, 44, 19]
let sorted = selectionSort(data)
assert(sorted == data.sorted())
print("[Swift SelectionSort] Selection sort verified: \(sorted)")
